### ISSUE TITLE
[Snackbar] Message text is cut off or not visible at all for multiline (maxActionInlineWidth)

### ISSUE THREAD

**Issue by daniel-tailored** [material-components/material-components-android#1876]
Hi,
we came across this problem working on accessibility for our app.

**Problem**
afaik the problem is caused in `SnackbarContentLayout` by the logic in `onMeasure` regarding `updateViewsWithinLayout`. 
I'm sure you are aware but just for the sake of completeness:
First the message text is multiline so the layout changes to vertical, but after that it is re-evaluated and since the text has now the full width available it is single line again and therefore the layout changes back to horizontal. This leads to cutting off the message text or being not visible at all.

It might sound like an edge case at first but actually with font and display size in place (try out the largest option for both) this scenario can happen quite easily (especially in other languages) and frankly happens to us without trying to fabricate something.

**Example**
Pixel 4XL, Android 11, Material Lib 1.2.1 (same outcome for 1.3.0-alpha03)
font size: largest
display size: largest

`AppCompatActivity` with `Theme.MaterialComponents.Light.NoActionBar`

```kotlin
Snackbar.make(binding.coordinatorLayout, "Hey, you are offline.", Snackbar.LENGTH_INDEFINITE)
   .setMaxInlineActionWidth(resources.getDimensionPixelSize(R.dimen.design_snackbar_action_inline_max_width))
    .setAction("Please try again"){}
    .show()
```

![image](https://user-images.githubusercontent.com/62700024/98559922-7c95b180-22a7-11eb-8c2b-2f40f990fe74.png)

chross check, message text that always is multiline

```kotlin
Snackbar.make(binding.coordinatorLayout, "Hey, you are offline or have a bad connection.", Snackbar.LENGTH_INDEFINITE) 
   .setMaxInlineActionWidth(resources.getDimensionPixelSize(R.dimen.design_snackbar_action_inline_max_width))
   .setAction("Please try again"){}
   .show()
```

![image](https://user-images.githubusercontent.com/62700024/98563005-f7140080-22aa-11eb-9376-8f789e684745.png)

Would appreciate to hear from you since we need this to to provide a safe and accessible way to use our app for everyone.

I guess it is similar (or caused by the same bug) to this [issue](https://github.com/material-components/material-components-android/issues/781) but it also seems it might got lost (with closing google issue tracker and no one assigned) or at least no update since 2019.

Please let me know if I can provide anything.

Thx in advance.

**Comment by drchen** [comment]
Hi, I will take a look into the issue soon and I'm going to close the issue as duplicate to #781. : )

**Comment by liweiyap** [comment]
The default value of `R.dimen.design_snackbar_action_inline_max_width` is 128 dp.

What if we have our own custom max inline action width that is < 128 dp?
```java
Snackbar.setMaxInlineActionWidth(some_value_less_than_128);
```

Another [alternative](https://stackoverflow.com/a/54177574/12367873) might be to set the attribute `maxActionInlineWidth` directly to `@dimen/design_snackbar_action_inline_max_width` in our main theme.

Both approaches have so far been viable for me; since coming to issues like this for help, I haven't yet had a problem with forcing the action onto a new line. For reference, I also used the largest font size (emboldened) and the largest display size on my device (model: Samsung A32).

**Comment by drchen** [comment]
Thanks for the info. It looks like we have another bug to fix. Will work on it.

**Comment by drchen** [comment]
I think we should prefer multiline in this corner case - the message can be fit in one line in the multiline layout, but cannot in single-line layout. I'll create a change and confirm with the team about the behavior.

**Comment by daniel-tailored** [comment]
> Thanks for the info. It looks like we have another bug to fix. Will work on it.

@drchen not sure if the reply was towards me or the issue I posted.. anyways tried again with `rc01` and still the same problem :/
thx for looking into this and making the material components accessible.

![image](https://user-images.githubusercontent.com/62700024/148935142-00797fcb-ead2-4dc3-b7d0-8205598aeeb1.png)
![image](https://user-images.githubusercontent.com/62700024/148935200-3d63d9fd-b0e5-4a5a-890e-a8535cf758b3.png)
![image](https://user-images.githubusercontent.com/62700024/148935278-25aa9fff-b45a-4667-b750-bef97ccb901f.png)

**Comment by drchen** [comment]
The fix was submitted several days ago so it's not released yet. 

It will probably be released with 1.6 alpha versions I guess.

**Comment by daniel-tailored** [comment]
> The fix was submitted several days ago so it's not released yet.
> 
> It will probably be released with 1.6 alpha versions I guess.

@drchen ah sry just saw the commit is not included in that release 🙈 
thx for the reply anyways, since I wasn't sure if it was towards my initial issue or the different one brought forward from @liweiyap 

will check the next release/alpha then, hopefully it resolves the a11y problems 🤞 
thx.
